nntrainer/tensor/tensor.cpp

```diff
@@ -3418,6 +3418,20 @@ Tensor &Tensor::erf(Tensor &out) const {
   return out;
 }
 
+void Tensor::sin_transform(Tensor &out) {
+  if (getDataType() == ml::train::TensorDim::DataType::FP32) {
+    sine_transformation(this->size(), getData<float>(), out.getData<float>());
```

Comment:
```suggestion
void Tensor::sin_transform(Tensor &out, float alpha) {
  if (getDataType() == ml::train::TensorDim::DataType::FP32) {
    sine_transformation(this->size(), getData<float>(), out.getData<float>(), alpha);
```

having an extra parameter `float alpha` seems reasonable since currently alpha is fixed to 1.0. (or will alpha always be 1.0?)

Reply:
alpha in trigonometric calculation is only used for rotary embedding (RoPE)
In other cases, it will be zero

Reply:
But in a second thought, there is no reason for not having it as well.
I applied the change